lib/components/ButtonGroupView.jsx

```diff
@@ -53,100 +51,103 @@ const MruMenuItems = (mruFiles, openFile) => {
-                    <Glyphicon glyph="folder-open" />Add a .hex file
-                </Dropdown.Toggle>
-                <Dropdown.Menu>
-                    { mruMenuItems }
-                    <MenuItem divider />
-                    <MenuItem onSelect={openFileDialog}>Browse...</MenuItem>
-                </Dropdown.Menu>
-            </Dropdown>
-            <Button onClick={refreshAllFiles}>
-                <Glyphicon glyph="refresh" />Reload .hex files
-            </Button>
-            <Button onClick={closeFiles}>
-                <Glyphicon glyph="minus-sign" />Clear files
-            </Button>
-
-            <hr style={{ borderColor: 'transparent', margin: '5px 0px' }} />
-
-            <Button
-                onClick={performRecover}
-                disabled={!targetIsReady || !targetIsRecoverable}
-            >
-                <Glyphicon glyph="remove-sign" />Erase all
-            </Button>
-            <Button
-                onClick={performWrite}
-                disabled={!targetIsReady || !targetIsWritable}
-            >
-                <Glyphicon glyph="download-alt" />Write
-            </Button>
-            <Button
-                onClick={performRecoverAndWrite}
-                disabled={!targetIsReady || !targetIsRecoverable}
-            >
-                <Glyphicon glyph="save" />Erase all & write
-            </Button>
-        </div>
+        <Button
+            onClick={performWrite}
+            disabled={!targetIsReady || !targetIsWritable}
+        >
+            <Glyphicon glyph="download-alt" />Write
+        </Button>
     );
 };
 
+const ButtonGroupView = ({
+    openFile,
+    closeFiles,
+    refreshAllFiles,
+    openFileDialog,
+    onToggleFileList,
+    performJLinkWrite,
+    performUSBSDFUWrite,
+    performRecover,
+    performRecoverAndWrite,
+    mruFiles,
+    targetType,
+    targetIsReady,
+    targetIsWritable,
+    targetIsRecoverable,
+}) => (
+    <div>
+        <Dropdown pullRight id="files-dropdown">
+            <Dropdown.Toggle onClick={onToggleFileList}>
+                <Glyphicon glyph="folder-open" />Add a .hex file
+            </Dropdown.Toggle>
+            <Dropdown.Menu>
+                {MruMenuItems(mruFiles, openFile)}
+                <MenuItem divider />
+                <MenuItem onSelect={openFileDialog}>Browse...</MenuItem>
+            </Dropdown.Menu>
+        </Dropdown>
+        <Button onClick={refreshAllFiles}>
+            <Glyphicon glyph="refresh" />Reload .hex files
+        </Button>
+        <Button onClick={closeFiles}>
+            <Glyphicon glyph="minus-sign" />Clear files
+        </Button>
+
+        <hr style={{ borderColor: 'transparent', margin: '5px 0px' }} />
+
+        <Button
+            onClick={performRecover}
+            disabled={!targetIsReady || !targetIsRecoverable}
+        >
+            <Glyphicon glyph="remove-sign" />Erase all
+        </Button>
+        {WriteButton(
```

Comment:
I think it would be simpler to change the function with less footprint and better keeping the react structure:
```js
const getWriteAction = (targetType, performJLinkWrite, performUSBSDFUWrite) => {
    if (targetType === CommunicationType.JLINK) {
         return performJLinkWrite;
    }
    if (targetType === CommunicationType.USBSDFU) {
        return performUSBSDFUWrite;
    }
}
```
...
```jsx
<Button
    onClick={getWriteAction(targetType, performJLinkWrite, performUSBDSDFUWrite)}
    disabled={!targetIsReady || !targetIsWritable}
>
    <Glyphicon glyph="download-alt" />Write
</Button>
```